modules/videoModule/index.js

```diff
@@ -88,7 +89,20 @@ export function PbVideo(videoCore_, getConfig_, pbGlobal_, pbEvents_, videoEvent
     const adUrl = bid.vastUrl;
     options.adXml = bid.vastXml;
     options.winner = bid.bidder;
-    loadAdTag(adUrl, divId, options);
+
+    const { useLocal } = getConfig('cache') || {};
+
+    // in case of using local cache, VAST should be prefetched
+    // needed for make blob always reachable, regardless of player context
+    if (useLocal && !options.prefetchedVastXml) {
+      fetch(adUrl)
+        .then(response => response.text())
+        .then(prefetchedVastXml => loadAd(adUrl, divId, {...options, prefetchedVastXml}))
+        .catch((err) => logError(`Unable to fetch vast for bid ${bid.bidder}`, err))
+      return;
+    }
+
```

Comment:
Why are these changes necessary ? I don't think this is needed. This function is to render a given Bid in a video player, it has nothing to do with caching. The cache is never used here, why fetch the vast xml ?

Reply:
So what you're saying is even if there is no GAM or Ad Server, we always cache the bid locally as long as `cache.useLocal` is defined ? In that case I think this flow makes sense, though in practice this probably won't be a common use case since there is technically no need to cache a bid if there's no ad server involved

Reply:
it does feel like wasted work because `options.adXml` would be exactly the same as `options.prefetchedVastXml`

Reply:
I think a publisher would (or should know to) only configure `config.cache` when using an ad server for secondary auction; i can't think of any other practical use case for caching